### PR TITLE
Add combo box component for Microsoft Graph applications

### PR DESCRIPTION
**Pull request by jansorg**
This is part of #5190 and commits a part of the code to simplify the code review.

It adds a combo box component which displays Microsoft Graph applications. 

The new component is added to the `azure-intellij-resource-connector-aad` sub-project.

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Combobox listing Azure AD applications.
+ */
+class AzureApplicationComboBox extends AzureComboBox<Application> {
+    private final Supplier<List<Application>> supplier;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java:19]: no need to add this field, 
you can use `get/setItemsLoader` of `AzureComboBox` directly.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    protected String getItemText(Object item) {
+        if (item instanceof Application) {
+            return ((Application) item).displayName;
+        }
+        return super.getItemText(item);
+    }
+
+    @NotNull
+    @Override
+    protected List<? extends Application> loadItems() throws Exception {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java:44]: this method can be removed if you use the provided `setItemsLoader`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java (hunk 1 above)
Comment (jansorg) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java]: Thank you, that's a good point! I missed the superclass's item loader. I've updated this PR to use it instead.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java (hunk 2 above)
Comment (jansorg) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureApplicationComboBox.java]: Fixed as well.

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**